src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -118,7 +118,8 @@
      (format "sweep of stale reports (threshold: %s)"
              (format-period report-ttl))
      (jdbc/with-transacted-connection db
-       (scf-store/delete-reports-older-than! (ago report-ttl))))
+       (scf-store/delete-reports-older-than! (ago report-ttl))
+       (scf-store/delete-old-unassociated-catalogs! (ago report-ttl))))
```

Comment:
When we update pe-ext, we'll need to change the 'don't pull reports that would be ttl-ed' check to look at catalogs as well.